app/services/bulk_hosts_manager.rb

```diff
@@ -38,4 +38,17 @@ def rebuild_configuration
     end
     all_fails
   end
+
+  def assign_taxonomy(taxonomy, optimistic_import)
```

Comment:
Can you add documentation on what an optimistic import is? I'm always a fan of [yardoc](https://yardoc.org/) parameters. So something like:
```suggestion
  # @param [Boolean] optimistic_import
  #   A description of the parameter here
  def assign_taxonomy(taxonomy, optimistic_import)
```